pkg/udf/map_udf.go

```diff
@@ -157,28 +160,49 @@ func (u *MapUDFProcessor) Start(ctx context.Context) error {
 		}()
 
 	} else {
+		// As a vertex can be either Map mode or Batch mode we reuse the same server-info file and socket file for both
 		// Wait for server info to be ready
 		serverInfo, err := sdkserverinfo.SDKServerInfo(sdkserverinfo.WithServerInfoFilePath(sdkclient.MapServerInfoFile))
 		if err != nil {
 			return err
 		}
 
-		mapClient, err := mapper.New(serverInfo, sdkclient.WithMaxMessageSize(maxMessageSize))
-		if err != nil {
-			return fmt.Errorf("failed to create map client, %w", err)
-		}
-		mapHandler = rpc.NewUDSgRPCBasedMap(vertexName, mapClient)
-
-		// Readiness check
-		if err := mapHandler.WaitUntilReady(ctx); err != nil {
-			return fmt.Errorf("failed on map UDF readiness check, %w", err)
-		}
-		defer func() {
-			err = mapHandler.CloseConn(ctx)
+		// if Batch Map mode is enabled create the client and handler for that accordingly
+		if enableBatchMapUdf {
```

Comment:
this will also go away, right?

Reply:
Not right now, in this PR we would not unify all the map functionalities. Would be better to do in a follow up PR as a separate task

Reply:
Let's do it via follow up